DependencyInjection/SonataAdminExtension.php

```diff
@@ -214,13 +214,47 @@ public function prepend(ContainerBuilder $container)
     {
         $bundles = $container->getParameter('kernel.bundles');
 
-        if (isset($bundles['JMSDiExtraBundle'])) {
+        if (!isset($bundles['JMSDiExtraBundle'])) {
+            return;
+        }
+
+        $configs = $container->getExtensionConfig($this->getAlias());
+        $config = $this->processConfiguration(new Configuration(), $configs);
+        if (!$config['options']['enable_jms_di_extra_autoregistration']) {
+            return;
+        }
+
+        $sonataAdminPattern = 'Sonata\AdminBundle\Annotation';
+        $annotationPatternsConfigured = false;
+
+        $diExtraConfigs = $container->getExtensionConfig('jms_di_extra');
+        foreach ($diExtraConfigs as $diExtraConfig) {
+            if (isset($diExtraConfig['annotation_patterns'])) {
+                $annotationPatternsConfigured = true;
+                break;
+            }
+        }
+
+        if ($annotationPatternsConfigured) {
+            $container->prependExtensionConfig(
+                'jms_di_extra',
+                array(
+                    'annotation_patterns' => array($sonataAdminPattern),
+                )
+            );
+        } else {
```

Comment:
I have moved the prependExtensionConfig outside the if instead, I think it's more clear this way that in both cases a prepend gets done, but with different values.